Packs/CrowdStrikeMalquery/Integrations/CrowdStrikeMalquery/CrowdStrikeMalquery.py

```diff
@@ -0,0 +1,425 @@
+        params = {'ids': request_id}
+        return self.http_request(method="GET", url_suffix='/entities/samples-fetch/v1', headers=headers, params=params,
+                                 resp_type="response")
+
+    def get_files_metadata(self, files_ids):
+        params = {'ids': files_ids}
+        return self.http_request(method="GET", url_suffix='/entities/metadata/v1', params=params)
+
+
+def test_module(client: Client, args: dict):
+    """
+    Returning 'ok' indicates that an access token was obtained successfully. Connection to the service is successful.
+
+    Args:
+        client:  Client
+
+    Returns:
+        'ok' if test passed, anything else will fail the test.
+    """
+    try:
+        client.get_access_token()
+    except Exception as e:
+        raise DemistoException(
+            f"Test failed. Please check your parameters. \n {e}")
+    return 'ok'
+
+
+def exact_search_command(client: Client, args: dict) -> CommandResults:
+    pattern_names = ['hex', 'ascii', 'wide']
+    patterns = [
+        {
+            "type": key,
+            "value": args[key]
+        } for key in pattern_names if args.get(key)
+    ]
+
+    # must provide a pattern (hex, ascii ot wide string)
+    if not patterns:
+        raise DemistoException("You must provide a query to search in one of the following patterns: Hex, ASCII, "
+                               "Wide string")
+
+    # dates format: YYYY/MM/DD
+    query_filters = {
+        "limit": int(args.get('limit', '')) if args.get('limit') else None,
+        "filter_meta": argToList(args.get('filter_meta')),
+        "filter_filetypes": argToList(args.get('file_types')),
+        "max_size": args.get('max_size'),
+        "min_size": args.get('min_size'),
+        "max_date": args.get('max_date'),
+        "min_date": args.get('min_date'), }
+    options = remove_None_values_keys(query_filters)
+    body = {"options": options, "patterns": patterns}
+    raw_response = client.exact_search(body)
+    entry_context = {"Request_ID": raw_response.get('meta', {}).get('reqid')}
+
+    human_readable = tableToMarkdown('Search Result', entry_context, removeNull=True)
+
+    return CommandResults(
+        readable_output=human_readable,
+        outputs_prefix='Malquery',
+        outputs_key_field='Request_ID',
+        outputs=entry_context,
+        raw_response=raw_response)
+
+
+def fuzzy_search_command(client: Client, args: dict) -> CommandResults:
+    pattern_names = ['hex', 'ascii', 'wide']
+    patterns = [
+        {
+            "type": key,
+            "value": args[key]
+        } for key in pattern_names if args.get(key)
+    ]
+    # must provide a pattern (hex, ascii ot wide string)
+    if not patterns:
+        raise DemistoException("You must provide a query to search in the following patterns: Hex, ASCII, Wide string")
+
+    query_filters = {
+        "limit": int(args.get('limit', '')) if args.get('limit') else None,
```

Comment:
you can decide on a different default value.
is there an option for pagination?
```suggestion
        "limit": int(args.get('limit', '100')),
```

Reply:
@yaakovi  I looked at the response model in the swagger file and there is an option for pagination but on the actual response there wasn't a `pagination` field as supposed and it also didn't appear on the malquery documentation.